include/envoy/network/dns.h

```diff
@@ -41,11 +41,14 @@ class DnsResolver {
   /**
    * Initiate an async DNS resolution.
    * @param dns_name supplies the DNS name to lookup.
+   * @param dns_lookup_ip_version the DNS IP version lookup policy.
    * @param callback supplies the callback to invoke when the resolution is complete.
    * @return if non-null, a handle that can be used to cancel the resolution.
    *         This is only valid until the invocation of callback or ~DnsResolver().
    */
-  virtual ActiveDnsQuery* resolve(const std::string& dns_name, ResolveCb callback) PURE;
+  virtual ActiveDnsQuery* resolve(const std::string& dns_name,
+                                  const std::string& dns_lookup_ip_version,
```

Comment:
Instead of passing the dns_lookup_ip_version as a string, consider resolving it into an enum when you process the configuration and use that enum here. This will be cheaper than doing the string comparisons every time you resolve.